File: Springboot_API/RetaCantabria/src/main/java/org/example/DTO/CalendarioDTO.java
package org.example.DTO;

import com.fasterxml.jackson.annotation.JsonFormat;
import org.example.Entidades.Calendario;

import java.time.LocalDate;
import java.time.LocalDateTime;

public record CalendarioDTO(
        String fecha,
        String detalles,
        String recomendaciones,
        Long idRuta,
        Long idUsuario

) {
    public CalendarioDTO(Calendario calendario) {
        this(
                calendario.getFecha().toString(),
                calendario.getDetalles(),
                calendario.getRecomendaciones(),
                calendario.getRutasIdruta().getIdRuta(),
                calendario.getUsuarioIdusuario().getIdUsuario()
        );
    }
}

File: Aplicación Escritorio/ModeloTDO/CalendarioDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ModeloDTO
{
    public class CalendarioDTO
    {
        public String fecha { get; set; }
        public String detalles { get; set; }
        public String recomendaciones { get; set; }
        public long idRuta { get; set; }
        public long idUsuario { get; set; }

    }
}
